src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -189,3 +189,49 @@ function evaluate(bn::BayesianNetwork)
     end
     return evaluation_env, logp
 end
+
+function evaluate_with_values(bn::BayesianNetwork, parameter_values::AbstractVector, model)
+    # Use the exact same node ordering as BUGSModel
+    bugsmodel_node_order = model.flattened_graph_node_data.sorted_nodes
+
+    var_lengths = model.transformed_var_lengths
+    evaluation_env = deepcopy(model.evaluation_env)
+    current_idx = 1
+    logprior, loglikelihood = 0.0, 0.0
+
+    for vn in bugsmodel_node_order
+        # Get the corresponding index in BayesianNetwork
+        i = findfirst(name -> name == vn, bn.names)
```

Comment:
this is pretty bad in terms of complexity, we have `names_to_ids` as a field for `BayesianNet`, it should work?